### ISSUE TITLE
gixsql: consider support for ocesql compatible datasrc and optional dbtype

### ISSUE THREAD

**Issue by GitMensch** [mridoni/gix#71]
The documentation clearly shows the correct format: `<dbtype>://<host>[:port][/dbname][?[opt1=val1]&...]`

I suggest two adjustments:

* dbtype optional - no need to specify `<dbtype>://` if there is no dbtype assume `gixsql` (which I assume would mean "use the default type", which - if only one is driver is available should be this one, otherwise a compile-time constant definition (which could be set also during configure `--with-default=´]
* at least for PostgreSQLsupport the ocesql format https://github.com/opensourcecobol/Open-COBOL-ESQL/blob/44ba99e752880e4ab75d089143e86d7901f194f5/dblib/ocesql.c#L518-L535: `[dbname][@host][:port]`

Note: Feel free to decline these suggestions, but I think the first would make sense in general and the second would ease migrations from ocesql.

**Comment by GitMensch** [comment]
There were changes in the regex with 1.0.11 but I'm not sure - is the dbtype optional already?
What is used when the dbtype is "gixsql"?

Do you consider an ocesql compatibility also for the connection string (=datasrc), maybe this should already work?

**Comment by mridoni** [comment]
> There were changes in the regex with 1.0.11 but I'm not sure - is the dbtype optional already? What is used when the dbtype is "gixsql"?
> 
> Do you consider an ocesql compatibility also for the connection string (=datasrc), maybe this should already work?

No, this just fixes the problem with the client encoding (all parameters, actually). I am leaving the other connection string modifications things for 1.1.12 (if I can) or soon after that.
